fdbclient/S3BlobStore.actor.cpp

```diff
@@ -312,7 +312,7 @@ Reference<S3BlobStoreEndpoint> S3BlobStoreEndpoint::fromString(const std::string
 			// The parameter is known to S3BlobStoreEndpoint so it must be numeric and valid.
 			char* valueEnd;
 			int ivalue = strtol(value.toString().c_str(), &valueEnd, 10);
-			if (*valueEnd || (ivalue == 0 && value.toString() != "0"))
+			if (valueEnd != nullptr || (ivalue == 0 && value.toString() != "0"))
```

Comment:
from https://man7.org/linux/man-pages/man3/strtol.3.html i have below, it seems that if `valueEnd` is NULL, it does not mean the conversion has failed.


```
    If endptr is not NULL, strtol() stores the address of the first
       invalid character in *endptr.  If there were no digits at all,
       strtol() stores the original value of nptr in *endptr (and
       returns 0).  In particular, if *nptr is not '\0' but **endptr is
       '\0' on return, the entire string is valid.
```

Reply:
to clarify, it seems the only way `valueEnd == nullptr` is that `valueEnd` is initialized as `valueEnd` and passed into `strtol`, so it is just not possible here.

Reply:
> it seems that if valueEnd is NULL, it does not mean the conversion has failed.

Right. That's checked by `ivalue == 0 && value.toString() != "0"` condition.
`valueEnd != nullptr` is definitely a conversion error.

Reply:
`valueEnd` will always be not `nullptr`. So I'll just keep the `value.toString()` in a temporary variable to solve the problem.